### ISSUE TITLE
examples/video: add a WebM video

### ISSUE THREAD

**Issue by hajimehoshi** [hajimehoshi/ebiten#3065]
### Operating System

- [ ] Windows
- [ ] macOS
- [ ] Linux
- [ ] FreeBSD
- [ ] OpenBSD
- [ ] Android
- [ ] iOS
- [ ] Nintendo Switch
- [ ] PlayStation 5
- [ ] Xbox
- [ ] Web Browsers

### What feature would you like to be added?

We want to make a proof of concept to play a WebM video.

https://github.com/xlab/libvpx-go is available. In the long run, we have to develop a pure Go WebM decoder.

### Why is this needed?

Now we can play MPEG videos but the file size tends to be large compared to WebM videos.

**Comment by TotallyGamerJet** [comment]
WebM is just the container for VP8, VP9, and AV1. There is already a reader written in Go for webm. You could also use the x/image/vp8 package to actually decode the video frames. Hopefully this can be a good starting point

https://github.com/ebml-go/webm
https://pkg.go.dev/golang.org/x/image/vp8

**Comment by hajimehoshi** [comment]
Thanks, but VP9, AV1, and Opus are not supported in a pure Go way yet, right?

**Comment by TotallyGamerJet** [comment]
Correct. I just tried to use the vp8 package but i couldn't get it to work. Kept giving me Golden / Alt frames are not implemented. So it seems the package is not feature complete

**Comment by TotallyGamerJet** [comment]
Actually there is an opus package but idk if it works or not. 

https://github.com/pion/opus

**Comment by realskyquest** [comment]
Is the api for webm player gonna be based on the mpeg and will it replace the use of mpeg.

I support the change from mpeg to webm, it's mature and is common format in web, it will definitely reduce size.

This will be a complicated new feature.

**Comment by hajimehoshi** [comment]
I don't plan to add a feature for WebM to Ebitengine itself so far. I just plan to update the example to use WebM, even though this uses Cgo.

**Comment by hajimehoshi** [comment]
By the way, there are some projects to use Wasm and Wazero to run C libraries in the pure Go way:

https://github.com/wasilibs
https://github.com/ncruces/go-sqlite3

**Comment by realskyquest** [comment]
Well, wazero works fine and it's tested on production environment. I guess gen2brain only has webp, the support for webm have to be done using libwebm.

I remember reading this blog about using wazero, they did appreciate the crossplatform capabilities in exchange for performance hit.

https://www.dolthub.com/blog/2023-05-19-wasm-in-go/
https://github.com/gen2brain/webp
https://github.com/webmproject/libwebm
